http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java

```diff
@@ -40,9 +37,15 @@ public static ConnectivityController build(final GameListing gameListing) {
       keys = {KeyPart.IP},
       rates = {@Rate(limit = 10, duration = 1, timeUnit = TimeUnit.MINUTES)})
   @RolesAllowed(UserRole.HOST)
-  public boolean checkConnectivity(
+  public Response checkConnectivity(
       @Auth final AuthenticatedUser authenticatedUser, final String gameId) {
-    return connectivityCheck.test(
-        gameListing.getHostForGame(authenticatedUser.getApiKey(), gameId));
+
+    if (!connectivityCheck.gameExists(authenticatedUser.getApiKey(), gameId)) {
+      return Response.status(400).build();
```

Comment:
I'm always a bit suspicious when I see explicit 400 status codes, they always more or less hint on syntax errors.
I think 404 might be a better choice here actually.

See https://httpstatuses.com/400 and https://httpstatuses.com/404

Reply:
404 potentially has issues:
- 'resource' generally means the URL resource (right?)
- we won't be able to distinguish between an actual 404 and 404 due to bad data. For example, a common mistake is not register a controller with the ServerApplication. In such cases you get a 404 as the path you created is not actually listening.

You're right though about generic 400 being more syntax error @RoiEXLab .

422 - unprocessable entity seems to fit the bill.

Reply:
Updated to 422: 27e3415469f2cd47c20d788387de73b7c1cab4f4